### PR TITLE
feat(lang): add OCaml

### PR DESCRIPTION
**Pull request by Ben10164**
## Description

Adds an extra that adds language support for ocaml.

Adds LSP and completions. Fairly simple/small config.

## Related Issue(s)

None

## Screenshots

lsp ocaml

## Checklist

- [x] I've read the [CONTRIBUTING](https://github.com/LazyVim/LazyVim/blob/main/CONTRIBUTING.md) guidelines.

### CODE REVIEW

File: lua/lazyvim/plugins/extras/lang/ocaml.lua [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+              "*.opam",
+              "esy.json",
+              "package.json",
+              ".git",
+              "dune-project",
+              "dune-workspace",
+              "*.ml"
+            )(fname)
+          end,
+          on_attach = on_attach,
```
Comment (dpetka2001) [inline lua/lazyvim/plugins/extras/lang/ocaml.lua:35]:
```suggestion

```
Not needed I believe?

File: lua/lazyvim/plugins/extras/lang/ocaml.lua [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    end,
+  },
+  {
+    "neovim/nvim-lspconfig",
+    opts = {
+      servers = {
+        ocamllsp = {
+          get_language_id = function(_, ftype)
+            return ftype
+          end,
```
Comment (dpetka2001) [inline lua/lazyvim/plugins/extras/lang/ocaml.lua:23]: Are you sure this is correct? In [nvim-lspconfig ocamllsp docs](https://github.com/neovim/nvim-lspconfig/blob/master/lua/lspconfig/server_configurations/ocamllsp.lua) it seems to return the `language_id` of the `ftype`, not the `ftype` itself. Is this actually needed?

Comment (Ben10164) [reply lua/lazyvim/plugins/extras/lang/ocaml.lua]: Good catch!

**Comment by folke** [comment]
ty!

**Pull request merged**